src/families/tron/bridge/js.js

```diff
@@ -375,6 +375,7 @@ const getAccountShape = async (info, syncConfig) => {
           operations.length > 0
             ? operations[operations.length - 1].date
             : new Date(),
+        swapHistory: [],
       };
```

Comment:
this should be preserving the potential `maybeExistingSubAccount` the same way we preserve pending operations

Reply:
😕 I dont really get it

Reply:
See the code

```js
        pendingOperations: maybeExistingSubAccount
          ? maybeExistingSubAccount.pendingOperations
          : [],
```

we didn't do `[]` because in fact we always recreate the sub account reference in Tron case,

this is why there is this comment too

```
  // FIXME: this is bad for perf: we should reconciliate with potential existing data
  // we need to KEEP REF as much as possible & use minimalOperationsBuilderSync
```

so it's really similar for your swapHistory usage, if you set it to [] and do not take into account the `maybeExistingSubAccount` you are going to lose the swaps history.

Reply:
Gotcha, I think
```
swapHistory: maybeExistingSubAccount
   ? maybeExistingSubAccount.swapHistory
   : [],
```